scalalib/src/ScalaModule.scala

```diff
@@ -185,6 +187,12 @@ trait ScalaModule extends JavaModule { outer =>
       )
   }
 
+  override def docSources: Sources = T.sources {
+    // Scaladoc 3.0.0 is consuming tasty files
+    if(isScala3(scalaVersion() && !isScala3Milestone(scalaVersion())) Seq((compile().classes))
+    else allSources()
+  }
```

Comment:
Fixing compile issue

```suggestion
  override def docSources: Sources = T.sources {
    // Scaladoc 3.0.0 is consuming tasty files
    if(isScala3(scalaVersion()) && !isScala3Milestone(scalaVersion())) Seq(compile().classes)
    else allSources()
  }
```